### PR TITLE
add accvlab video decode backend

### PR DESCRIPTION
**Pull request by xupinjie**
Fixes #

Changes proposed in this pull request:
- Add ACCV-Lab video decoding backend in DataLoader.

## Before submitting

- [x] I've read and followed all steps in the [Making a pull request](https://github.com/NVIDIA/Isaac-GR00T/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

### CODE REVIEW

**Comment by xupinjie** [comment]
We attempted to implement on-demand decoding for a subset of frames using [ACCV-Lab](https://github.com/NVIDIA/ACCV-Lab), a GPU video decoding tool. This optimization successfully reduced the training time for 1,200 iterations from 48 minutes to 24 minutes.

The training configuration is detailed below:

- shard-size: 512
- global-batch-size: 256
- num-gpus: 8x A100
- episode_sampling_rate: 0.1
- data.delta_indices:[0]

The attached image illustrates the comparison of time and loss curves between our proposed method (NVC, on-demand-decode) and the original baseline (TorchCodec with full-frames-deocde).

training_comparison_20260118_224102_gpu8_bs256_shard512 (1)

**Comment by xupinjie** [comment]
## install ACCV-Lab using GPU video training

https://github.com/NVIDIA/ACCV-Lab/blob/main/packages/on_demand_video_decoder/docs/installation.md

### prepare ffmpeg
```bash
#install from source code
wget https://ffmpeg.org/releases/ffmpeg-4.4.6.tar.xz
tar -xf ffmpeg-4.4.6.tar.xz

cd ffmpeg-4.4.6
LDSOFLAGS=-Wl,-rpath,\''$$$$ORIGIN'\'  ./configure --enable-shared --disable-encoders --disable-decoders --enable-decoder=vp9 --arch=x86_64 --prefix=/PATH/TO/YOUR/FFMPEG

make -j
make install

export FFMPEG_DIR=/PATH/TO/YOUR/FFMPEG
```

### install ACCV-Lab
```bash
git clone --recursive git@github.com:NVIDIA/ACCV-Lab.git
cd ACCV-Lab/

cd build_config  # ← Assuming you are in the root directory of the ACCV-Lab repository
pip install --no-build-isolation .
cd ../

cd packages/on_demand_video_decoder   # ← Assuming you are in the root directory of the ACCV-Lab repository
pip install --no-build-isolation .


### test if install successful
python samples/SampleDecodeFromGopFiles.py 
```

## train GR00T
Base on the original training scripts, add:
`--video-backend "nvc"`
And
`--multiprocessing-context "spawn"`

**Comment by mingxin-zheng** [comment]
Thanks @xupinjie .

The following dockerfile seems able to build gr00t with the accvlab support:

```dockerfile
FROM nvcr.io/nvidia/pytorch:25.04-py3

ENV NVIDIA_DRIVER_CAPABILITIES=graphics,utility,compute,video

RUN apt-get update && apt-get install -y \
    build-essential yasm cmake libtool git pkg-config \
    libass-dev libfreetype6-dev libvorbis-dev \
    autoconf automake texinfo tmux wget \
    ffmpeg libavcodec-dev libavformat-dev libavutil-dev libswscale-dev

# Build ffmpeg from source with VP9 decoder support for ACCV-Lab GPU video training
ENV FFMPEG_DIR=/opt/ffmpeg
ENV LD_LIBRARY_PATH=/opt/ffmpeg/lib:${LD_LIBRARY_PATH}
ENV PATH=/opt/ffmpeg/bin:${PATH}
RUN cd /tmp && \
    wget https://ffmpeg.org/releases/ffmpeg-4.4.6.tar.xz && \
    tar -xf ffmpeg-4.4.6.tar.xz && \
    cd ffmpeg-4.4.6 && \
    LDSOFLAGS=-Wl,-rpath,\''$$ORIGIN'\' ./configure \
        --enable-shared \
        --disable-encoders \
        --disable-decoders \
        --enable-decoder=vp9 \
        --arch=x86_64 \
        --prefix=${FFMPEG_DIR} && \
    make -j$(nproc) && \
    make install && \
    rm -rf /tmp/ffmpeg-4.4.6 /tmp/ffmpeg-4.4.6.tar.xz

# Clone ACCV-Lab (will install after venv is created)
RUN pip install scikit-build cmake ninja
RUN cd /opt && git clone --recursive https://github.com/NVIDIA/ACCV-Lab.git

RUN pip install --upgrade pip setuptools
RUN pip install uv
RUN sed -i s/dill==0.3.9/dill==0.3.8/g /etc/pip/constraint.txt
RUN sed -i s/scipy==1.15.2/scipy==1.15.3/g /etc/pip/constraint.txt
RUN pip install imageio h5py boto3 transformers[torch] deepspeed timm peft diffusers wandb tianshou dm_tree openai albumentations==1.4.18 decord==0.6.0 torchcodec==0.4.0
RUN mkdir -p /opt
RUN cd /opt && git clone https://github.com/facebookresearch/pytorch3d.git
RUN cd /opt/pytorch3d && pip install .

COPY src/gr00t/ /workspace/gr00t/
RUN cd /workspace/gr00t && uv sync && uv pip install -e .

# Install ACCV-Lab into the venv for GPU video decoding
# First install build dependencies into the venv
RUN uv pip install --python /workspace/gr00t/.venv/bin/python scikit-build cmake ninja
# Set CMAKE_POLICY_VERSION_MINIMUM to work around old pybind11 CMakeLists.txt
ENV CMAKE_POLICY_VERSION_MINIMUM=3.5
RUN cd /opt/ACCV-Lab/build_config && \
    uv pip install --no-build-isolation --python /workspace/gr00t/.venv/bin/python . && \
    cd ../packages/on_demand_video_decoder && \
    uv pip install --no-build-isolation --python /workspace/gr00t/.venv/bin/python .

RUN apt-get install -y libegl1
RUN mkdir -p /usr/share/glvnd/egl_vendor.d && \
    cat >/usr/share/glvnd/egl_vendor.d/10_nvidia.json <<'EOF'
{
    "file_format_version" : "1.0.0",
    "ICD" : {
        "library_path" : "libEGL_nvidia.so.0"
    }
}
EOF

RUN mkdir -p /usr/share/vulkan/icd.d && \
    cat >/usr/share/vulkan/icd.d/nvidia_icd.json <<'EOF'
{
    "file_format_version": "1.0.0",
    "ICD": {
        "library_path": "libGLX_nvidia.so.0",
        "api_version": "1.2.140"
    }
}
EOF
```
